spec/features/season_items_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'ProduceItems', type: :feature do
+  let!(:produce_item) { ProduceItem.create(name: 'carambola', category: 'fruit') }
```

Comment:
Since we use it in the `before` block, there's no need to use `let!`. `before` will make sure we have this item by the time we create `ItemSeason`

Reply:
But in the `let!` we are creating a `produce_item`, and in the `before` we create an `item_season` using that `produce_item.id`. If I remove it the tests break.

Reply:
I thought you meant completely removing it. Leaving it as below works:
`RSpec.describe 'ProduceItems', type: :feature do
  let(:produce_item) { ProduceItem.create(name: 'carambola', category: 'fruit') }

  before { ItemSeason.create(produce_item_id: produce_item.id, month_index: 1, country_code: 'br') }`
  
So the advantage would be that let is lazy-evaluated: it is not evaluated until the first time
the method it defines is invoked, while let! is invoked before each example?

Reply:
Yeap, exactly. As a rule of thumb, avoid eager evaluation unless you really need it.